projects/dotcms-js/src/lib/core/core-web.service.mock.ts

```diff
@@ -1,29 +1,100 @@
+    HttpHeaders
+} from '@angular/common/http';
+import { RequestMethod, RequestOptionsArgs } from '@angular/http';
+import { RequestOptionsParams } from 'dotcms-js';
 
 @Injectable()
 export class CoreWebServiceMock {
+    constructor(private _http: HttpClient) {}
 
-    constructor(private _http: Http) {}
+    request(options: RequestOptionsArgs): Observable<any> {
+        const optionsArgs: RequestOptionsParams = {
+            params: new HttpParams()
+        };
 
-    request(options: any): Observable<any> {
-        return this._http.request(new Request(options));
+        if (options.params) {
+            Object.keys(options.params).forEach((key) => {
+                optionsArgs.params = optionsArgs.params.set(key, options.params[key]);
+            });
+        }
+
+        return this._http
+            .request(
+                new HttpRequest(RequestMethod[options.method], options.url, options.body, {
+                    params: optionsArgs.params
+                })
+            )
+            .pipe(
+                filter(<T>(event: HttpEvent<T>) => event.type === HttpEventType.Response),
+                map((resp: HttpResponse<any>) => {
+                    try {
+                        return resp.body;
+                    } catch (error) {
+                        return resp;
+                    }
+                })
+            );
     }
 
-    public requestView(options: RequestOptionsArgs): Observable<ResponseView> {
-        return this._http.request(new Request(<any>options)).pipe(
-            map((resp) => {
-                return new ResponseView(resp);
-            })
-        );
+    requestView(options: RequestOptionsArgs): Observable<ResponseView> {
+        const optionsArgs: RequestOptionsParams = {
+            headers: new HttpHeaders(),
+            params: new HttpParams()
+        };
+
+        if (options.params) {
+            Object.keys(options.params).forEach((key: string) => {
+                optionsArgs.params = optionsArgs.params.set(key, options.params[key]);
+            });
+        }
+
+        if (options.search) {
+            optionsArgs.params = this.setHttpParams(options.search, optionsArgs.params);
+        }
+
+        return this._http
+            .request(
+                new HttpRequest(RequestMethod[options.method], options.url, options.body, {
+                    params: optionsArgs.params
+                })
+            )
+            .pipe(
+                filter(<T>(event: HttpEvent<T>) => event.type === HttpEventType.Response),
+                map(<T>(resp: HttpResponse<T>) => {
+                    return new ResponseView(resp);
+                })
+            );
     }
 
-    public subscribeTo(httpErrorCode: number): Observable<any> {
+    subscribeTo(httpErrorCode: number): Observable<any> {
         return of({
             error: httpErrorCode
         });
     }
+
+    private setHttpParams(urlParams: any, httpParams: HttpParams): HttpParams {
```

Comment:
```suggestion
    private setHttpParams(urlParams: ?, httpParams: HttpParams): HttpParams {
```

I think Angular have a type for this.